checker/tests/guieffect/Java8Lambdas.java

```diff
@@ -0,0 +1,176 @@
+        // is not an @UI PolymorphicLambdaRunner ? Or, failing that (call.invalid.ui) since doEither is @PolyUIEffect ?
+        safePolymorphicLambdaRunner.doEither(e -> e.dangerous());
+        safePolymorphicLambdaRunner.doEither(
+                new @UI PolymorphicFunctionalInterface<UIElement>() {
+                    public void executePolymorphic(UIElement arg) {
+                        arg.dangerous();
+                    }
+                });
+        @UI PolymorphicLambdaRunner uiPolymorphicLambdaRunner = new @UI PolymorphicLambdaRunner(elem);
+        //:: error: (call.invalid.ui)
+        uiPolymorphicLambdaRunner.doEither(
+                e -> e.repaint()); // Safe at runtime, but not by the type system!
+        //:: error: (call.invalid.ui)
+        uiPolymorphicLambdaRunner.doEither(e -> e.dangerous());
+        PolymorphicFunctionalInterface<UIElement> func1 = e -> e.repaint();
+        // No error, why? //:: error: (assignment.type.incompatible)
+        PolymorphicFunctionalInterface<UIElement> func2 = e -> e.dangerous(); // Incompatible types!
+        //:: error: (assignment.type.incompatible)
+        PolymorphicFunctionalInterface<UIElement> func2p =
+                (new @UI PolymorphicFunctionalInterface<UIElement>() {
+                    public void executePolymorphic(UIElement arg) {
+                        arg.dangerous();
+                    }
+                });
+        @UI PolymorphicFunctionalInterface<UIElement> func3 = e -> e.dangerous();
+        safePolymorphicLambdaRunner.doEither(func1);
+        safePolymorphicLambdaRunner.doEither(func2);
+        //:: error: (call.invalid.ui)
+        uiPolymorphicLambdaRunner.doEither(func1);
+        //:: error: (call.invalid.ui)
+        uiPolymorphicLambdaRunner.doEither(func2);
+        //:: error: (call.invalid.ui)
+        uiPolymorphicLambdaRunner.doEither(func3);
+    }
+
+    @UIEffect
+    public static void uiContextTestCases(UIElement elem) {
+        LambdaRunner runner = new LambdaRunner(elem);
+        //:: error: (call.invalid.ui)
+        runner.doSafe(e -> e.dangerous());
+        runner.doUI(e -> e.repaint());
+        runner.doUI(e -> e.dangerous());
+        PolymorphicLambdaRunner safePolymorphicLambdaRunner = new PolymorphicLambdaRunner(elem);
+        // No error, why? :: error: (argument.type.incompatible)
+        safePolymorphicLambdaRunner.doEither(e -> e.dangerous());
+        @UI PolymorphicLambdaRunner uiPolymorphicLambdaRunner = new @UI PolymorphicLambdaRunner(elem);
+        uiPolymorphicLambdaRunner.doEither(e -> e.dangerous());
+    }
+
+    public @UI PolymorphicFunctionalInterface<UIElement> returnLambdasTest1() {
+        return e -> e.dangerous();
+    }
+
+    // This should be an error without an @UI annotation on the return type. No?
+    public PolymorphicFunctionalInterface<UIElement> returnLambdasTest2() {
+        // No error, why? :: error: (return.type.incompatible)
```

Comment:
Another test that should be failing.

Reply:
This would also be explained by the ```super.visitX``` invocation preceding the inference of the lambda inference, because again that's where the subtyping checks to generate that error would occur.